ShippingApp/src/main/java/com/javaschool/queue/QueueSender.java

```diff
@@ -0,0 +1,105 @@
+package com.javaschool.queue;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.javaschool.common.PackageSize;
+import com.javaschool.common.PackageType;
+import com.javaschool.common.QueueException;
+import com.javaschool.common.QueueMessageRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.amqp.core.AmqpTemplate;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class QueueSender {
+
+    private static final Logger logger = LoggerFactory.getLogger(QueueSender.class);
+
+    private AmqpTemplate rabbitTemplate;
+
+    @Value("${config-rabbit.routing-key}")
+    public String ROUTING_KEY_SHIPPING;
+
+    @Value("${config-rabbit.exchange}")
+    public String EXCHANGE_SHIPPING;
+
+    public QueueSender(final AmqpTemplate rabbitTemplate) {
+        this.rabbitTemplate = rabbitTemplate;
+    }
+
+    private String sendMessage(String queueMessage) {
+        String message = (String) rabbitTemplate.convertSendAndReceive(EXCHANGE_SHIPPING, ROUTING_KEY_SHIPPING, queueMessage);
+        logger.info(message);
+        return message;
+    }
+
+    private String messageRequest(String type){
+        ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
do you need a new ObjectMapper each time you send a message?

Reply:
It was modified to create only one instance and not create one each time a message was sent.